### ISSUE TITLE
Assert::numeric() with string will always evaluate to true

### ISSUE THREAD

**Issue by julienfalque** [phpstan/phpstan-webmozart-assert#8]
When calling `Webmozart\Assert\Assert::numeric()` with a string variable, the following error is reported:
```
Call to static method Webmozart\Assert\Assert::numeric() with string will always evaluate to true.
```

As a string *may* be a numeric value, the error is unexpected.

This also occurs with a hardcoded non-numeric argument, i.e. `Assert::numeric('foo')`:

```
Call to static method Webmozart\Assert\Assert::numeric() with 'foo' will always evaluate to true.
```

Here the error is expected as the value is hardcoded, though the message is wrong: it should be _always false_ instead of _always true_.

The error is not reported anymore when calling `is_numeric()` directly:
```php
if (!is_numeric($value)) {
    throw new \InvalidArgumentException();
}
```
